packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -1118,6 +1135,32 @@ class _ScaffoldLayout extends MultiChildLayoutDelegate {
 
       final double xOffset = hasCustomWidth ? (size.width - snackBarWidth!) / 2 : 0.0;
       positionChild(_ScaffoldSlot.snackBar, Offset(xOffset, snackBarYOffsetBase - snackBarSize.height));
+
+      assert((){
+        // Whether a floating SnackBar has been offsetted too high.
+        //
+        // To improve the developper experience, this assert is done after the call to positionChild.
+        // if we assert sooner the SnackBar is visible because its defaults position is (0,0) and
+        // it can cause confusion to the user as the error message states that the SnackBar is off screen.
+        if (isSnackBarFloating) {
+          final bool snackBarVisible = (snackBarYOffsetBase - snackBarSize.height) > 0;
```

Comment:
```suggestion
          final bool snackBarVisible = (snackBarYOffsetBase - snackBarSize.height) >= 0;
```

Should this be >= ? Wouldn't a position at the top edge be valid?

Reply:
Good catch! I pushed a fix and updated the related test to check several SnackBar positions (fully offscreen, partially offscreen, right on the top). Thanks!.